### PR TITLE
Adding uberenv support for Gitlab CI

### LINKED ISSUE #473: Leverage RADIUSS Shared CI in Caliper
Per [RADIUSS Shared CI documentation](https://radiuss-shared-ci.readthedocs.io/en/latest/sphinx/user_guide/index.html), here is a curated list of tasks to perform in order to leverage the shared CI infrastructure:

- [x] Use uberenv to build Caliper dependencies and generate a CMake Cache file:
  - [x] Setup Uberenv
  - [x] Setup Radiuss-spack-configs
  - [x] Build Caliper with Uberenv
  - [x] Duplicate Caliper Spack package in Radiuss-spack-configs and update the implementation to use CachedCMakePackage

- [x] Create a build-and-test script that takes a spack spec an input, the script must then:
  - [x] Use uberenv to build the dependencies
  - [x] Retrieve the generated CMake Cache file and use it to build Caliper and the tests
  - [x] Test Caliper

> RAJA has an example build-and-test script [here](https://github.com/LLNL/RAJA/blob/develop/scripts/gitlab/build_and_test.sh).

- [x] Setup the CI to use RADIUSS Shared CI.
  - [x] We will prefer to have the mirroring up and running at this stage.

### PR DESCRIPTION
**Pull request by garretthooten**
resolves #473

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,53 +1,74 @@
...
-    build_config: "quartz-intel"
-  extends: .build-quartz
+# We define the following GitLab pipeline variables:
+variables:
+# Required information about GitHub repository
+  GITHUB_PROJECT_NAME: "Caliper"
+  GITHUB_PROJECT_ORG: "LLNL"
+# Use an LLNL service user to run CI. This prevents from running pipelines as
+# an actual user.
+  LLNL_SERVICE_USER: ""
```
[the remark this answers is not in the document]
Comment (daboehme) [reply .gitlab-ci.yml:31]: Caliper is it's own thing, so yes we should set up a service user. Do you have some instructions where/how I can do that?

File: scripts/gitlab/build-and-test.sh [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    if [[ "${option}" != "--build-only" ]]
+    then # don't run the tests that are known to fail
+        date
+        ctest --output-on-failure -T test 2>&1 | tee tests_output.txt
+        date
+    else #run all tests like normal
+        date
+        ctest --output-on-failure -T test 2>&1 | tee tests_output.txt
+        date
+    fi
```
Comment (adrienbernede) [inline scripts/gitlab/build-and-test.sh:155]:
```suggestion
    date
    ctest --output-on-failure -T test 2>&1 | tee tests_output.txt
    date

```

**Review by adrienbernede — APPROVED** [review]

**Comment by adrienbernede** [comment]
@daboehme Ready to go. (Execution time reduced to ~7 minutes 🎉 )

**Comment by adrienbernede** [comment]
@daboehme anything holding you from merging ?

**Comment by daboehme** [comment]
> @daboehme anything holding you from merging ?

No, I think we're good to go. Happy to see this complete! Thanks @garretthooten and @adrienbernede!

**Pull request merged**